linear_algebra/src/power_iteration.py

```diff
@@ -42,19 +44,19 @@ def power_iteration(
     # Set convergence to False. Will define convergence when we exceed max_iterations
     # or when we have small changes from one iteration to next.
 
-    convergence = False
-    lamda_previous = 0
-    iterations = 0
-    error = 1e12
+    convergence: bool = False
+    lamda_previous: int = 0
+    iterations: int = 0
+    error: float = 1e12
```

Comment:
Type hints are not really required on local variables that are set to literals.  They create a lot of cognitive clutter.